apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.tsx

```diff
@@ -272,24 +272,19 @@ router.post(
         parsedBody.self_enrollment_use_enrollment_code,
         false,
       );
-      const selfEnrollmentBeforeDateEnabled = propertyValueWithDefault(
-        courseInstanceInfo.selfEnrollment?.beforeDateEnabled,
-        parsedBody.self_enrollment_enabled_before_date_enabled,
-        false,
-        { isUIBoolean: true },
-      );
 
       const selfEnrollmentBeforeDate = propertyValueWithDefault(
         parseDateTime(courseInstanceInfo.selfEnrollment?.beforeDate ?? ''),
-        parseDateTime(parsedBody.self_enrollment_enabled_before_date),
+        // We'll only serialize the value if self-enrollment is enabled.
+        parsedBody.self_enrollment_enabled && parsedBody.self_enrollment_enabled_before_date
```

Comment:
should we be posting this? I would think we just would POST undefined if not checked.